src/key_gen/tests.rs

```diff
@@ -109,50 +110,88 @@ fn all_nodes_being_responsive() -> Result<()> {
 #[test]
 fn having_max_unresponsive_nodes_still_work() -> Result<()> {
     let mut rng = rand::thread_rng();
-    let mut non_responsives = BTreeSet::<u64>::new();
-    for i in 0..(NODENUM - THRESHOLD - 1) as u64 {
-        let _ = non_responsives.insert(i);
+    let mut all_nodes = BTreeSet::<u64>::new();
+    for i in 0..NODENUM as u64 {
```

Comment:
can we do something like:
let all_nodes : BTreeSet::<u64> = (0..NODENUM).map(|idx| idx as u64).collect();

Reply:
ninja'd